src/domain/mongoose_domain_sup.erl

```diff
@@ -0,0 +1,19 @@
+-module(mongoose_domain_sup).
```

Comment:
It would be nice to have some wrapper interfaces for adding/removing children for this supervisor. direct usage of `supervisor` module has a strong spaghetti code taste

Reply:
wait, no. better completely remove dynamic adding/removal of the children for this supervisor.